sphinx/writers/latex.py

```diff
@@ -2003,6 +2003,7 @@ def visit_index(self, node, scre=re.compile(r';\s*')):
                     self.body.append(r'\index{%s|see{%s}}' % (p1, p2))
                 else:
                     logger.warning('unknown index entry type %s found', type)
+                self.body.append('%\n')
```

Comment:
This causes blank line when text paragraph starts afterwards. This blank line may facilitate a page break and then the indexed page number will be off by one. For this reason I am considering using

    self.body.append('\ignorespaces ')

Another approach is to insert something like `\ifhmode\hskip-1pt\hskip1pt\relax\fi` before the first `\index`. This is actually extracted from comments in the LaTeX source code! (whow). But its is complicated and its only purpose is to force a non-zero `\lastskip` which triggers the `\index` command to issue the `\ignorespaces`. Thus, maybe easier to output ourselves the `\ignorespaces ` systematically.

Reply:
Please read

    self.body.append('\\ignorespaces ')

sorry, got rusty during summer ...

Reply:
The reason we at Sphinx can use `\ignorespaces` is that in inline mode, the `:index:` in rst source must have a space before. Thus there is a space before `\index` in output. Hence the `\ignorespaces` after does not cause two successive words to touch.

Reply:
Hmm... I really had a too long vacation. When using the ````:index:`foo` ```` role, the latex will contain `\index{foo}foo` and there is no space after `\index{foo}` at all anyway. What I wanted to say I guess is that Sphinx never produces `foo\index{bar} bar` type of output currently, so `foo\index{bar}\ignorespaces  bar` which would typeset in PDF `foobar` will never happen if we use `\ignorespaces ` extra.

I think 

    self.body.append('\\ignorespaces ')

is the way to go.